src/index.js

```diff
@@ -376,19 +293,98 @@ class TemplateContentScript extends ContentScript {
     }
     await this.sendToPilot({ bills })
   }
+
+  async checkIfLoggedAtStart() {
+    this.log('info', 'First authentication check')
+    const isHidden = document
+      .querySelector('#idEspaceClientDivMobileConnected')
+      .getAttribute('class')
+      .includes('u-hide')
+    if (isHidden) {
+      return false
+    } else {
+      return true
+    }
+  }
+
+  async checkWelcomeMessage() {
+    this.log('info', 'checkWelcomeMessage starts')
+    if (
+      document.querySelector('.c-headerCelUser__name').textContent.length > 0 &&
+      document.querySelector('.contrat-en-cours').textContent.length > 0
+    )
+      return true
+    else return false
+  }
+
+  async checkIfFullfilled() {
+    function sortTruthy(value) {
+      return value.length > 0
+    }
+    const neededInfos = [
+      document.querySelector('#idEmailContact_Infos').textContent,
+      document.querySelector('#ProfilConsulterAdresseFacturation_nomComplet')
+        .textContent,
+      document.querySelector('#ProfilConsulterAdresseFacturation_adresse')
+        .textContent,
+      document.querySelector(
+        '#ProfilConsulterAdresseFacturation_complementAdresse'
+      ).textContent,
+      document.querySelector('#ProfilConsulterAdresseFacturation_commune')
+        .textContent,
+      document.querySelector('#idNumerosTelephone_Infos').textContent
+    ]
+    const truthyInfos = neededInfos.filter(sortTruthy)
+
+    if (truthyInfos.length === neededInfos.length) {
+      return true
+    }
+    return false
+  }
+
+  async checkActiveSession() {
+    this.log('debug', 'Starting checkActiveSession')
+    const formIsInvisible = document
+      .querySelector('#view-mode-non-connecte')
+      .getAttribute('class')
+      .includes('u-hide')
+    if (formIsInvisible) {
+      return true
+    } else {
+      return false
+    }
+  }
+
+  async handleForm(credentials) {
+    this.log('debug', 'Starting handleForm')
+    const loginElement = document.querySelector('input[id="email"]')
+    const passwordElement = document.querySelector('input[id="motdepasse"]')
+    const submitButton = document.querySelector('button[id="login-btn"]')
+
+    loginElement.value = credentials.login
+    passwordElement.value = credentials.password
+    if (loginElement.value.length > 0 && passwordElement.value.length > 0) {
+      this.log('info', 'Login and password fullfilled')
+      submitButton.click()
+    } else {
+      this.log('info', 'something went wrong while filling values')
```

Comment:
What happens if the autologin fails? Shouldn't we display the login form to the user ?

Reply:
You right. I'll add some boolean returns as the `authWithCredentials` method will run `waitForUserAuthentication` if anything's wrong.

Reply:
Test had been added 👍